src/io.jl

```diff
@@ -215,7 +215,7 @@ promote_type2(::Type{Null}, ::Type{WeakRefString{UInt8}}) = Union{WeakRefString{
 promote_type2(::Type{Any}, ::Type{Null}) = Null
 promote_type2(::Type{Null}, ::Type{Null}) = Null
 
-function detecttype(io, opt::CSV.Options{D}, prevT) where {D}
+function detecttype(io, opt::CSV.Options{D}, prevT, seen) where {D}
```

Comment:
I'd rather call this `uniques` or `levels` (`seen` is used in `Base.unique` for a set indicating whether a value was seen). Same for `categoricals` above (using the same name for both would be clearer).

Reply:
updated